spec/Section 4 -- Composition.md

```diff
@@ -115,6 +115,656 @@ type User {
+- For each {schema} in {schemas}:
+  - Let {rootSubscription} be the root mutation type defined in {schema}, if it
+    exists.
+  - Let {namedSubscriptionType} be the type with the name `Subscription` in
+    {schema}, if it exists.
+  - If {rootSubscription} is defined:
+    - {rootSubscription} must be named `Subscription`.
+  - Otherwise, {namedSubscriptionType} must not be defined.
+
+**Explanatory Text**
+
+This rule enforces that, for any subgraph schema, if a root subscription type is
+defined, it must be named `Subscription`. Defining a root subscription type with
+a name other than `Subscription` or using a differently named type alongside a
+type explicitly named `Subscription` creates inconsistencies in schema design
+and violates the composite schema specification.
+
+**Examples**
+
+Valid Example:
+
+```graphql example
+# Subgraph A
+schema {
+  subscription: Subscription
+}
+
+type Subscription {
+  productCreated: Product
+}
+
+type Product {
+  id: ID!
+  name: String
+}
+```
+
+The following counter-example violates the rule because `RootSubscription` is
+used as the root subscription type, but a type named `Subscription` is also
+defined.
+
+```graphql counter-example
+# Subgraph A
+schema {
+  subscription: RootSubscription
+}
+
+type RootSubscription {
+  productCreated: Product
+}
+
+type Subscription {
+  deprecatedField: String
+}
+```
+
+### Key Fields Select Invalid Type
+
+**Error Code**
+
+`KEY_FIELDS_SELECT_INVALID_TYPE`
+
+**Severity**
+
+ERROR
+
+**Formal Specification**
+
+- Let {schema} be the set of all source schemas.
+  - Let {types} be the set of all object types that are annotated with the
+    `@key` directive in {schema}.
+  - For each {type} in {types}:
+    - Let {keyFields} be the set of fields referenced by the `fields` argument
+      of the `@key` directive on {type}.
+    - For each {field} in {keyFields}:
+      - Let {fieldType} be the type of {field}.
+      - {fieldType} must not be a `List`, `Interface`, or `Union` type.
+
+**Explanatory Text**
+
+The `@key` directive is used to define the set of fields that uniquely identify
+an entity. These fields must reference scalars or object types to ensure a valid
+and consistent representation of the entity across subgraphs. Fields of types
+`List`, `Interface`, or `Union` cannot be part of a `@key` because they do not
+have a well-defined unique value.
+
+**Examples**
+
+In this valid example, the `Product` type has a valid `@key` directive
+referencing the scalar field `sku`.
+
+```graphql example
+type Product @key(fields: "sku") {
+  sku: String!
+  name: String
+}
+```
+
+In the following counter-example, the `Product` type has an invalid `@key`
+directive referencing a field (`featuredItem`) whose type is an interface,
+violating the rule.
+
+```graphql counter-example
+type Product @key(fields: "featuredItem { id }") {
+  featuredItem: Node!
+  sku: String!
+}
+
+interface Node {
+  id: ID!
+}
+```
```

Comment:
Should this be looking at top-level fields like `featuredItem`, or only at the "leaf" fields like `id`?